src/shared-components/dropdown/index.tsx

```diff
@@ -59,10 +65,10 @@ export const Dropdown = ({
     event: React.ChangeEvent<HTMLSelectElement>,
   ) => {
     const { target } = event;
-    const { value: nextValue, selectedOptions } = target;
+    const { selectedIndex, selectedOptions } = target;
+    const selectedOption = options[selectedIndex];
     if (selectedOptions && selectedOptions.length) {
-      const { label } = selectedOptions[0];
-      onChange({ value: nextValue, label });
+      onChange(selectedOption);
```

Comment:
I also changed this: it did not make sense to me why we were returning the `label` with the change, but in any case now we're returning the whole option object, including `value` and `label`, so that usage can do whatever it wants with this. 

I have an example of simplifying some code in a consumer app in another PR I will be tagging you in, where we were mapping and transforming an object and removing properties, which led to convoluted types: an initial state value was the whole object, but with `onChange` it turned into a `Partial` property. The new code is simpler.